scripts/sync_fofb/scan_dcc_links.py

```diff
@@ -0,0 +1,101 @@
+#! /usr/bin/python3
+
+''' Script to scan FOFB DCC FMC/P2P links.
+
+    This script reads DCC link partners PVs, builts a directed graph and saves
+    it on disk using json format.
+
+    Usage:  "./scan_dcc_links.py <JSON FILENAME> <LIST OF CRATES>", where
+
+            <JSON FILENAME>:    output filename
+            <LIST OF CRATES>:   list of crates' indexes separated by spaces
+
+    Author: Guilherme Ricioli
+'''
+
+import collections
+from epics import PV
+from itertools import product
+import json
+import sys
+
+import bpm_epics_ioc_slot_mapping
+
+# crates
+crates = [f"{int(i):02}" for i in sys.argv[2:]]
+
+# logical slots
+rtmlamp_slot = 3
+slots = [rtmlamp_slot, 13, 15, 17, 19]
+
+num_of_gts = 8
+link_partners_pvs_suffix = ["LinkPartnerCH" + f"{gt}" + "-Mon" for gt in range(num_of_gts)]
+
+def consume(iterator):
+	collections.deque(iterator, maxlen=0)
+
+global_pv_list = []
+def create_pv(name):
+	pv = PV(name)
+	global_pv_list.append(pv)
+	return pv
```

Comment:
I'd love to make this stuff into a separate python file so it can be imported and not repeated everywhere. Would you be ok with waiting for me to do that before merging?

Reply:
Of course!